configuration/src/main/java/io/airlift/configuration/validation/FileExists.java

```diff
@@ -0,0 +1,29 @@
+package io.airlift.configuration.validation;
+
+import javax.validation.Constraint;
+import javax.validation.Payload;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.ANNOTATION_TYPE;
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.ElementType.TYPE_USE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+@Target({ FIELD, METHOD, PARAMETER, ANNOTATION_TYPE, TYPE_USE })
+@Retention(RUNTIME)
+@Constraint(validatedBy = FileExistsValidator.class)
+@Documented
+public @interface FileExists
+{
+    // This message is ignored as FileExistsValidator builds constraint violation message
+    String message() default "file does not exist";
```

Comment:
We could make this empty and validate that it's empty inside the validator, in case someone changes it and expects it to work.